### PR TITLE
fixed #18254: Crash when switching between score and part with different measures selected

### LINKED ISSUE #18254: Crash when switching between score and part with different measures selected
### Issue type

Crash or freeze

### Steps to reproduce

1. Create score
2. Add some notes in 1st measure
3. Open part and select 1st measure 
4. Switch to score 
5. Add measure to the end
6. Switch several times between score and part

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/90187801/a5b5fb8e-940c-4d32-9a33-5065b1b085a3

### MuseScore Version

4.1 nightly master

### Regression

Yes, this used to work in **MuseScore 4.0.2**

### Operating system

Windows10

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #18254

### CODE REVIEW

File: src/inspector/models/abstractinspectormodel.cpp [hunk 1]
```diff
@@ -121,21 +121,13 @@ AbstractInspectorModel::AbstractInspectorModel(QObject* parent, IElementReposito
         return;
     }
 
-    setupCurrentNotationChangedConnection();
+    onCurrentNotationChanged();
```
Comment (cbjeukendrup) [inline src/inspector/models/abstractinspectormodel.cpp:124]: Isn't it bad that we are calling a virtual method inside the constructor?

File: src/inspector/models/inspectorlistmodel.cpp [hunk 2]
```diff
@@ -44,6 +44,8 @@ InspectorListModel::InspectorListModel(QObject* parent)
     listenSelectionChanged();
     context()->currentNotationChanged().onNotify(this, [this]() {
         listenSelectionChanged();
+
+        notfyModelsAboutNotationChanged();
```
Comment (RomanPudashkin) [inline src/inspector/models/inspectorlistmodel.cpp:48]: typo: no**ti**fy

File: src/inspector/models/inspectorlistmodel.cpp [hunk 3]
```diff
@@ -272,6 +281,13 @@ AbstractInspectorModel* InspectorListModel::modelBySectionType(InspectorSectionT
     return nullptr;
 }
 
+void InspectorListModel::notfyModelsAboutNotationChanged()
```
Comment (RomanPudashkin) [inline src/inspector/models/inspectorlistmodel.cpp:284]: Please add TRACEFUNC

**Review by RomanPudashkin — APPROVED** [review]

**Comment by DmitryArefiev** [comment]
Tested #18254 on Win10, Mac13, LinuxUbuntu 22.04 - **FIXED**

but found another one related to switching between score and part #18412 (also occurs in 4.1 builds)

**Pull request merged**